### PR TITLE
log: generate digest and log it in slow log (#9662)

### PR DESCRIPTION
**Pull request by lysu**
cherry-pick #9662 to 2.1

DO NOT merge until https://github.com/pingcap/parser/pull/279 done thx

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/10093)

### CODE REVIEW

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by crazycs520** [comment]
@jackysp PTAL

File: go.mod [hunk 1]
```diff
@@ -82,3 +82,5 @@ require (
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
 )
+
+replace github.com/pingcap/parser => github.com/lysu/parser v0.0.0-20190410025826-059b6f83c7bc
```
Comment (zimulala) [inline go.mod]: Do we need to update this?

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

File: go.mod (hunk 1 above)
Comment (lysu) [reply go.mod]: done~

**Comment by zz-jason** [comment]
@lysu please resolve conflicts.

**Comment by lysu** [comment]
/run-all-tests tidb-test=release-2.1 tikv=release-2.1 pd=release-2.1

**Pull request merged**